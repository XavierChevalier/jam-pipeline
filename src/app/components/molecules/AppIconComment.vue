<script setup lang="ts">
  import { mdiSpeechOutline } from '@/assets/images/icons/MdiExtra'
  import AppIcon from '@/app/components/atoms/AppIcon.vue'
  import { isPropertyValid } from '@/app/tools/component-properties/PropertyValidator'
  import { number } from 'yup'

  defineProps({
    showCounter: {
      type: Boolean,
      default: false,
    },
    counter: {
      type: Number,
      default: 0,
      validator: isPropertyValid(number().integer().min(0)),
    },
  })
</script>

<template>
  <span class="flex items-center text-neutral">
    <AppIcon :path="mdiSpeechOutline" />
    <span v-if="showCounter" class="ml-2">{{ counter }}</span>
  </span>
</template>
