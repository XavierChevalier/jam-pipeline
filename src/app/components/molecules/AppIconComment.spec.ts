import * as Stories from '@/app/components/molecules/AppIconComment.stories'
import { StorybookTestsGenerator } from '@/tests/tests-generators/StorybookTestsGenerator'

describe('AppIconComment', () => {
  StorybookTestsGenerator.fromStoriesExports(Stories)
    .snapshotEachStories()
    .property<number>('counter', (property) => {
      property
        .itShouldBeDefined()
        .itShouldHaveAValidatorFunction()
        .itShouldSuccessWith(0)
        .itShouldSuccessWith(10)
        .itShouldFailWith('negative number', -1)
    })
})
