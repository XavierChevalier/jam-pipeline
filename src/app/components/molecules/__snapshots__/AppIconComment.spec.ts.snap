// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppIconComment stories snapshots [snapshot] should render the "IconCommentWithCounter" component story correctly 1`] = `
<span
  class="flex items-center text-neutral"
>
  <anonymous-stub
    path="[object Object]"
  />
  <span
    class="ml-2"
  >
    10
  </span>
</span>
`;

exports[`AppIconComment stories snapshots [snapshot] should render the "JustIconComment" component story correctly 1`] = `
<span
  class="flex items-center text-neutral"
>
  <anonymous-stub
    path="[object Object]"
  />
  <!--v-if-->
</span>
`;
